satellite/analytics/service.go

```diff
@@ -101,8 +128,71 @@ func (service *Service) enqueueMessage(message segment.Message) {
 	}
 }
 
+func (service *Service) createHubSpotUser(ctx context.Context, data PayloadUser) (err error) {
+	if !service.config.Enabled {
+		return nil
+	}
+	defer mon.Task()(&ctx)(&err)
+	payloadBytes, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	body := bytes.NewReader(payloadBytes)
+
+	url := "https://api.hubapi.com/crm/v3/objects/contacts?hapikey=" + url.QueryEscape(service.config.HubSpotAPIKey)
+
+	req, err := http.NewRequestWithContext(ctx, "POST", url, body)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = errs.Combine(err, resp.Body.Close())
+	}()
+	return nil
+}
+
+func (service *Service) sendHubSpotEvent(ctx context.Context, dataEvent interface{}) (err error) {
+	if !service.config.Enabled {
+		return nil
+	}
+	defer mon.Task()(&ctx)(&err)
+	payloadBytes, err := json.Marshal(dataEvent)
+	if err != nil {
+		return err
+	}
+
+	body := bytes.NewReader(payloadBytes)
+
+	url := "https://api.hubapi.com/events/v3/send?hapikey=" + url.QueryEscape(service.config.HubSpotAPIKey)
+
+	req, err := http.NewRequestWithContext(ctx, "POST", url, body)
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		err = errs.Combine(err, resp.Body.Close())
+	}()
+	return nil
+}
+
 // TrackCreateUser sends an "Account Created" event to Segment.
-func (service *Service) TrackCreateUser(fields TrackCreateUserFields) {
+func (service *Service) TrackCreateUser(ctx context.Context, fields TrackCreateUserFields) {
 	traits := segment.NewTraits()
 	traits.SetName(fields.FullName)
 	traits.SetEmail(fields.Email)
```

Comment:
@mobyvb Is there a reason we have both segmentio and hubspot backends here? Shouldn't it be one or the other?

Reply:
@prerna-parashar will probably be able to give more information about this. But based on my understanding:
We currently are going from satellite -> segment -> hubspot for many of our analytics. I believe this is one step in a multi-step process to start sending analytics directly to hubspot. But I would still like to hear what Prerna says.

Reply:
@mobyvb,  @egonelbre Segment wasn't able to send behavioral events to HubSpot as we wanted so therefore I am directly hitting HubSpot Api along with sending requests to Segment to create user and sends events.

Satellite --> Segment, HubSpot